pkgs/by-name/pr/prismlauncher/package.nix

```diff
@@ -102,21 +106,29 @@ symlinkJoin {
       runtimeLibs =
         [
           # lwjgl
+          stdenv.cc.cc.lib
+          ## native versions
           glfw
-          libpulseaudio
-          libGL
           openal
-          stdenv.cc.cc.lib
 
-          vulkan-loader # VulkanMod's lwjgl
-
-          udev # oshi
+          ## openal
+          alsa-lib
+          libjack2
+          libpulseaudio
+          pipewire
+          portaudio
```

Comment:
I don't think that portaudio should be added, it increases the closure size by 90MB for no added benefit.

Reply:
I agree, PipeWire, Pulse and ALSA should be all we need realistically. If people want to use JACK, they can just add it using an override

Reply:
>  it increases the closure size by 90MB

This is misleading; Portaudio's *closure* size is ~90mb. According to `nix-tree /nix/store/99akjdz1gyi469ab0aqspg5j2rrh09mb-prismlauncher-8.4`, it's *added* size to this wrapper's closure is ~400kb. You can find a full explanation on how this number is determined/why it's important [here](https://github.com/utdemir/nix-tree/issues/14)

> for no added benefit

I can't really agree. There is most definitely a benefit in users having intended functionality available to them

> I agree, PipeWire, Pulse and ALSA should be all we need realistically

And while this may be true

> they can just add it using an override

I'd rather not give the few who do use JACK or Portaudio this answer in order to save ~2.4mb (libjack2's "added size" is ~2mb) in a 2.2GB closure. This is a minuscule amount compared to our 3 JREs -- or even Pipewire which adds ~270mb -- so I think it would be a bit strange to draw the line here and leave broke binaries

Reply:
PortAudio is completely redundant if yoh already have the other backends. It is just a portal audio library that uses ALSA, JACK, OSS or Pulse on Linux anyway. On macOS it just uses CoreAudio too, so it is really completely redundant if you already have support for other OpenAL backends.

I guess it's arguable if enough people use JACK in 2024 to warrant shipping it with Prism Launcher (I'd like to see prior precedent for this)

Adding support for all backends supported by OpenAL feels like a solution in search of a problem. I'd even say it's arguable if we should add PipeWire either, as most versions of the game will not even support PipeWire.

Reply:
IMO we can merge this by omitting portaudio, as it's redundant when all other backends are present.

Reply:
Yeah I think that would be a good compromise